src/withAxiom.ts

```diff
@@ -101,14 +102,40 @@ function withAxiomNextApiHandler(handler: NextApiHandler): NextApiHandler {
   };
 }
 
+export type AxiomRequest = NextRequest & { log: Logger };
+export type AxiomMiddleware = (
+  request: AxiomRequest,
+  event: NextFetchEvent
+) => NextMiddlewareResult | Promise<NextMiddlewareResult>;
+
 function withAxiomNextEdgeFunction(handler: NextMiddleware): NextMiddleware {
   return async (req, ev) => {
+    const report = {
+      request: {
+        ip: req.ip,
+        region: req.geo?.region,
+        host: req.nextUrl.host,
+        method: req.method,
+        path: req.nextUrl.pathname,
+        scheme: req.nextUrl.protocol.replace(':', ''),
+        statusCode: 0,
+        userAgent: req.headers.get('user-agent'),
+      },
+    };
+    const logger = log.with({
+      request: report.request,
+    });
+    const axiomRequest = req as AxiomRequest;
+    axiomRequest.log = logger;
+
     try {
-      const res = await handler(req, ev);
+      const res = await handler(axiomRequest, ev);
+      report.request.statusCode = res?.status || 0;
       ev.waitUntil(log.flush());
       return res;
     } catch (error) {
       log.error('Error in edge function', { error });
+      report.request.statusCode = 500;
```

Comment:
Will this do anything?

Reply:
probably would have been useful in case of EDGE_REPORT. I will remove it.

Reply:
We need to collect logs until flush in functions. Then we can attach the report (including the correct status code) to each log and send them all out.

Reply:
hmm this would have very different behavior than frontend I guess. how should we approach this? different logger class that doesn't throttle and send?

Reply:
I am also wondering if seeing the status of the request on each log is that useful. for checking the status of request, wouldn't one record of that be enough?